Framework/Muon/test/ApplyMuonDetectorGroupPairingTest.h

```diff
@@ -0,0 +1,631 @@
+  alg.setProperty("SpecifyGroupsManually", true);
+  alg.setProperty("PairName", "test");
+  alg.setProperty("Alpha", 1.0);
+  alg.setProperty("InputWorkspace", inputWSName);
+  alg.setProperty("InputWorkspaceGroup", wsGroupName);
+  alg.setProperty("Group1", "1-5");
+  alg.setProperty("Group2", "5-10");
+  alg.setProperty("TimeMin", 0.0);
+  alg.setProperty("TimeMax", 30.0);
+  alg.setProperty("RebinArgs", "");
+  alg.setProperty("TimeOffset", 0.0);
+  alg.setProperty("SummedPeriods", std::to_string(1));
+  alg.setProperty("SubtractedPeriods", "");
+  alg.setProperty("ApplyDeadTimeCorrection", false);
+  alg.setLogging(false);
+}
+
+// Set algorithm properties to sensible defaults (assuming data with 10 groups)
+// Use when entering workspaces to pair
+void setPairAlgorithmPropertiesForInputWorkspace(
+    ApplyMuonDetectorGroupPairing &alg, std::string inputWSName,
+    std::string wsGroupName) {
+  alg.setProperty("SpecifyGroupsManually", false);
+  alg.setProperty("PairName", "test");
+  alg.setProperty("Alpha", 1.0);
+  alg.setProperty("InputWorkspace", inputWSName);
+  alg.setProperty("InputWorkspaceGroup", wsGroupName);
+  alg.setLogging(false);
+}
+
+// Simple class to set up the ADS with the configuration required by the
+// algorithm (a MatrixWorkspace and an empty group).
+class setUpADSWithWorkspace {
+public:
+  setUpADSWithWorkspace(Workspace_sptr ws) {
+    AnalysisDataService::Instance().addOrReplace(inputWSName, ws);
+    wsGroup = boost::make_shared<WorkspaceGroup>();
+    AnalysisDataService::Instance().addOrReplace(groupWSName, wsGroup);
+  };
+
+  ~setUpADSWithWorkspace() { AnalysisDataService::Instance().clear(); };
+  WorkspaceGroup_sptr wsGroup;
+
+  const std::string inputWSName = "inputData";
+  const std::string groupWSName = "inputGroup";
+};
+
+} // namespace
+
+class ApplyMuonDetectorGroupPairingTest : public CxxTest::TestSuite {
+public:
+  // WorkflowAlgorithms do not appear in the FrameworkManager without this line
+  ApplyMuonDetectorGroupPairingTest() {
+    Mantid::API::FrameworkManager::Instance();
+  }
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static ApplyMuonDetectorGroupPairingTest *createSuite() {
+    return new ApplyMuonDetectorGroupPairingTest();
+  }
+  static void destroySuite(ApplyMuonDetectorGroupPairingTest *suite) {
+    delete suite;
+  }
+
+  void test_Init() {
+    ApplyMuonDetectorGroupPairing alg;
+    TS_ASSERT_THROWS_NOTHING(alg.initialize());
+    TS_ASSERT(alg.isInitialized());
+  }
+
+  void test_nonAlphanumericPairNamesNotAllowed() {
+    MatrixWorkspace_sptr ws = createAsymmetryWorkspace(10, 10);
+    setUpADSWithWorkspace setup(ws);
+
+    ApplyMuonDetectorGroupPairing alg;
+    alg.initialize();
+    setPairAlgorithmProperties(alg, setup.inputWSName, setup.groupWSName);
+
+    std::vector<std::string> badPairNames = {"",  "!", ";name;", ".",
```

Comment:
could we add `"42"` to the list of bad names (or some other number?

Reply:
Currently I am allowing names made from alphanumeric characters, so "42" is valid. This can be changed if we want

Reply:
ah ok. I misunderstood, I thought we had to have at least one letter and one number. Never mind